crates/q_cli/src/cli/chat/mod.rs

```diff
@@ -1632,6 +1803,94 @@ where
             }
         }
 
+        // Handle summarization completion if we were in summarization mode
+        if let Some(summarization_state) = self.summarization_state.take() {
+            // Get the latest message content (the summary)
+            let summary = match self.conversation_state.history_as_vec().last() {
+                Some(ChatMessage::AssistantResponseMessage(message)) => message.content.clone(),
+                _ => "Summary could not be generated.".to_string()
+            };
+            
+            // Store the summary in conversation_state
+            self.conversation_state.latest_summary = Some(summary.clone());
+            
+            // Clear the conversation but preserve the summary we just created
+            self.conversation_state.clear(true);
+            
+            // Create a special first assistant message that tells the user a summary is available
+            // Emphasize that the model will actively reference the summary
+            let special_message = AssistantResponseMessage {
+                message_id: None,
+                content: "Your conversation has been summarized and the history cleared. The summary contains the key points, tools used, code discussed, and insights from your previous conversation. I'll reference this summary when answering your future questions.".to_string(),
+                tool_uses: None,
+            };
+            
+            // Add the message
+            self.conversation_state.push_assistant_message(special_message);
+            
+            // Display confirmation message
+                
+            execute!(
+                self.output,
+                style::SetForegroundColor(Color::Green),
+                style::Print("\n✅ Conversation has been successfully summarized and cleared!\n\n"),
```

Comment:
```suggestion
                style::Print("\nConversation history has been compacted successfully!\n\n"),
```

Can we use a checkmark character with the color green rather than an emoji?